src/main/java/uk/ac/ebi/ampt2d/accession/common/generators/DecoratedGenerator.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.common.generators;
+
+import uk.ac.ebi.ampt2d.accession.common.accessioning.SaveResponse;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Generator that can encapsulate another generator with the purpose of decorating the output accession
+ *
+ * @param <MODEL>
+ * @param <DECORATED_ACCESSION>
+ * @param <ACCESSION>
+ */
+public class DecoratedGenerator<MODEL, DECORATED_ACCESSION, ACCESSION> implements
```

Comment:
Please leave this class out of the current PR so no new functionality is added. We will then create another ticket to support accession decoration.